libs/s25main/CampaingSettings.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2005 - 2021 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <array>
+#include <string>
+#include <vector>
+
+class CampaignSettings
+{
+public:
+    CampaignSettings(std::string filePath);
```

Comment:
If this is a path, make the argument a `bfs::path` not a string

And is it possible to pass/store the folder name here instead of the file name so that the path to the maps can be calculated inside the class not at some other places depending on the values inside the file?

It then might make sense to have a public static constant for the settings filename (e.g. for tests) so that we can change that when required.